File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int totalProduct = 1;
        int zeroCount = 0;

        for (int num : nums) {
            if (num == 0) {
                zeroCount++;
            }
            else {
                totalProduct *= num;
            }
        }

        vector<int> result(nums.size(), 0);
        for (int i = 0; i < nums.size(); i++) {
            if (zeroCount > 1) {
                result[i] = 0;
            }
            else if (zeroCount == 1) {
                result[i] = (nums[i] == 0) ? totalProduct : 0;
            }
            else {
                result[i] = totalProduct / nums[i];
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest_count = 0
        for num in nums:
            if (num - 1) not in num_set:
                current_num = num
                current_streak = 1

                while (current_num + 1) in num_set:
                    current_num += 1
                    current_streak += 1

                longest_count = max(longest_count, current_streak)
        
        return longest_count
                